AIO/src/org/aio/activities/quests/TheRestlessGhost.java

```diff
@@ -0,0 +1,190 @@
+                        isComplete = true;
+                    }
+                    break;
+                default:
+                    log("Unknown progress config value: " + getProgress());
+                    setFailed();
+                    break;
+            }
+        }
+    }
+
+    private void placeSkull() throws InterruptedException {
+        RS2Object coffin = getObjects().closest("Coffin");
+
+        if (!COFFIN.contains(myPosition())) {
+            getWalking().webWalk(COFFIN);
+        } else if(coffin != null && coffin.hasAction("Close") && getInventory().contains("Ghost's skull")) {
+            useSkull();
+        } else if (coffin != null && coffin.hasAction("Open")) {
+            openCoffin();
+        }
+    }
+
+    private void useSkull() throws InterruptedException {
+        if(getInventory().isItemSelected()) {
+            if(getObjects().closest("Coffin").interact("Use")){
+                Sleep.sleepUntil(() -> getProgress() != 4, 15000, 1500);
+            }
+        } else {
+            if(getInventory().interact("Use", "Ghost's skull")) {
+                Sleep.sleepUntil(() -> getInventory().isItemSelected(), 5000);
+            }
+        }
+    }
+
+    private void talkToGhost() throws InterruptedException {
+        NPC ghost = getNpcs().closest("Restless ghost");
+
+        if (!COFFIN.contains(myPosition())) {
+            getWalking().webWalk(COFFIN);
+        } else if (ghost == null) {
+            openCoffin();
+        } else if (canTalkToGhost() && (!getDialogues().inDialogue() || !myPlayer().isInteracting(ghost))) {
+            if (ghost.interact("Talk-to")) {
+                Sleep.sleepUntil(() -> getDialogues().inDialogue() && myPlayer().isInteracting(ghost), 5000);
+            }
+        } else {
+            getDialogues().completeDialogue(GHOST_OPTIONS);
+        }
+    }
+
+    private void openCoffin() throws InterruptedException {
+        RS2Object coffin = getObjects().closest("Coffin");
+
+        if (coffin != null && (coffin.interact("Search") || coffin.interact("Open"))) {
+            Sleep.sleepUntil(() -> getNpcs().closest("Restless ghost") != null, 10000, 1500);
+        }
+    }
+
+    private void searchAlter() throws InterruptedException {
+        RS2Object alter = getObjects().closest("Altar");
+
+        if (!WIZARD.contains(myPosition())) {
+            getWalking().webWalk(WIZARD);
+        } else if(alter != null && alter.interact("Search")) {
+            Sleep.sleepUntil(() -> getInventory().contains("Ghost's skull"), 5000);
+        }
+    }
+
+    private boolean canTalkToGhost() {
+        if(getInventory().contains("Ghostspeak amulet")) {
+            if(getInventory().interact("Wear", "Ghostspeak amulet")) {
+                Sleep.sleepUntil(() -> !getInventory().contains("Ghostspeak amulet"), 5000);
+            }
+        }
+        return getEquipment().getItemInSlot(EquipmentSlot.AMULET.slot).getName().equals("Ghostspeak amulet");
```

Comment:
This will break if the player is not wearing the amulet. I believe there is a function `isWearingItem` or something like that, which you should use

Reply:
When I tested this returned false if the player didn't wear that amulet, but I will look into other functions to use. But this worked and could return both true and false.